app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -182,6 +182,28 @@ def self.find_with_expiring_pools(_key, _operator, days_from_now)
           end
         end
 
+        def self.of_eos_index(release)
+          version, arch = case release
+                          when 'RHEL7'
+                            ['7', 'x86_64']
+                          when 'RHEL7 (POWER9)'
+                            ['7', 'ppc64le']
+                          when 'RHEL7 (ARM)'
+                            ['7', 'aarch64']
+                          when 'RHEL7 (System z (Structure A))'
+                            ['7', 's390x']
+                          when 'RHEL9', 'RHEL8', 'RHEL6', 'RHEL5'
+                            /RHEL(?<version>\d+)/ =~ release
+                            [version]
+                          else
+                            fail("Not supported EOS index: #{release}")
+                          end
+
+          query = joins(:operatingsystem).where("operatingsystems.name = ? and operatingsystems.major = ?", 'RedHat', version)
+          query = query.joins(:architecture).where("architectures.name = ?", arch) if arch.present?
+          query
```

Comment:
I think this can just rely on the host's stored status instead. You can just do a host search (`Host::Managed.search_for`) for approaching_end_of_maintenance, or one of the other values:

![image](https://github.com/Katello/katello/assets/22042343/6806c241-aa67-4bda-8d60-a6387c38c9f9)

This way you won't have to re-compute anything.